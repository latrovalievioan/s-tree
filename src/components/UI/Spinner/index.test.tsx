import { render, screen, cleanup } from '@testing-library/react';
import { Spinner } from '.';

describe('Modal', () => {
  beforeEach(() => {
    render(<Spinner width={10} height={10} border={2} skin="light" />);
  });

  afterEach(() => {
    cleanup();
  });

  it('Should render the component', () => {
    const spinner = screen.getByTestId('loading-spinner');

    expect(spinner).toBeInTheDocument();
    expect(spinner).toBeVisible();
  });

  it('The width, height and border of the component should be as passed', () => {
    const spinner = screen.getByTestId('loading-spinner');

    expect(spinner.style.width).toEqual('10px');
    expect(spinner.style.height).toEqual('10px');
    expect(spinner.style.borderWidth).toEqual('2px');
  });

  it('Should set the correct className for skin', () => {
    const spinner = screen.getByTestId('loading-spinner');

    expect(spinner.className).toContain('lightSkin');
  });

  it('Matches snapshot', () => {
    const spinner = screen.getByTestId('loading-spinner');

    expect(spinner).toMatchSnapshot();
  });
});
