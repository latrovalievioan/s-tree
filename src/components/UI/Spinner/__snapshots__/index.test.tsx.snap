// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > Matches snapshot 1`] = `
<div
  class="spinner lightSkin"
  data-testid="loading-spinner"
  style="width: 10px; height: 10px; border-width: 2px;"
/>
`;
